expose polynomial:power and cosine_with_restarts:num_cycles params

zetyquickly opened the pull request:
to get_scheduler func, 

* add it to the train_dreambooth.py

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/diffusers/pr_1737). All of your documentation changes will be reflected on that endpoint.

zetyquickly commented:
@pcuenca PR to an issue https://github.com/huggingface/diffusers/issues/1232

What's done:
* exposed only two parameters power and num_cycles on top of existing get_scheduler func API. it's possible to expose all of the parameters but taking into account that the issue wasn't there before we've noticed it, I assume it would pile up the code
* changed parameter num_cycles name of the cosine scheduler to not overlap with cosine_with_restarts, because they have different meanings
* added new parameters to the dreambooth's args

Haven't found any test script in test folder that tests optimizers.py

pcuenca commented on src/diffusers/optimization.py:
```diff
@@ -121,9 +121,9 @@ def get_cosine_schedule_with_warmup(
-            The number of waves in the cosine schedule (the defaults is to just decrease from the max value to 0
-            following a half-cosine).
+        num_periods (`float`, *optional*, defaults to 0.5):
+            The number of periods of the cosine function in a schedule (the defaults is to just decrease from the max
```
```suggestion
            The number of periods of the cosine function in a schedule (the default is to just decrease from the max
```

pcuenca reviewed (approved):
Looks good!

An alternative would be to use `scheduler_kwargs` in `get_scheduler`, but since we only have those two configurable schedulers I think it's good for now.

patrickvonplaten reviewed (approved):
Thanks @zetyquickly
